src/Microsoft.AspNet.Mvc.Core/ModelBinders/BodyModelBinder.cs

```diff
@@ -48,34 +47,21 @@ public BodyModelBinder()
                 return new ModelBindingResult(model: null, key: bindingContext.ModelName, isModelSet: false);
             }
 
-            object model = null;
             try
             {
-                model = await formatter.ReadAsync(formatterContext);
+                var model = await formatter.ReadAsync(formatterContext);
+
+                // key is empty to ensure that the model name is not used as a prefix for validation.
+                return new ModelBindingResult(model, key: string.Empty, isModelSet: true);
             }
             catch (Exception ex)
             {
-                model = GetDefaultValueForType(bindingContext.ModelType);
                 bindingContext.ModelState.AddModelError(bindingContext.ModelName, ex);
 
                 // This model binder is the only handler for the Body binding source.
                 // Always tell the model binding system to skip other model binders i.e. return non-null.
                 return new ModelBindingResult(model: null, key: bindingContext.ModelName, isModelSet: false);
```

Comment:
This was my bad. I think the first parameter here was supposed to be assigned the default value for type model calculated above?